app/controllers/concerns/current_zone_loader_decorator.rb

```diff
@@ -3,40 +3,28 @@
 CurrentZoneLoader.module_eval do
   extend ActiveSupport::Concern
 
-  def current_zone
-    return @current_zone if defined?(@current_zone)
+  def flow_zone
+    flow_io_zones = Spree::Zones::Product.active.where(
+      "meta -> 'flow_data' ->> 'country' = ?", ISO3166::Country[request_iso_code]&.alpha3
+    ).all.to_a
 
-    @current_zone = if (session_region_name = session['region']&.[]('name'))
-                      Spree::Zones::Product.find_by(name: session_region_name)
-                    elsif request_iso_code.present?
-                      flow_io_zones = Spree::Zones::Product.active.where(
-                          "meta -> 'flow_data' ->> 'country' = ?", ISO3166::Country[request_iso_code]&.alpha3
-                        ).all.to_a
-                      if flow_io_zones.present?
-                        request_ip = if Rails.env.production?
-                                       request.ip
-                                     else
-                                       Spree::Config[:debug_request_ip_address] || request.ip
-                                       # Germany ip: 85.214.132.117, Sweden ip: 62.20.0.196, Moldova ip: 89.41.76.29
-                                     end
-                        flow_io_session = FlowcommerceSpree::Session
-                                            .new(ip: request_ip, visitor: visitor_id_for_flow_io)
-                        # :create method will issue a request to flow.io. The experience, contained in the
-                        # response, will be available in the session object - flow_io_session.experience
-                        flow_io_session.create
-                        zone = Spree::Zones::Product.active.find_by(name: flow_io_session.experience&.key&.titleize)
-                        session['_f60_session'] = flow_io_session.id if zone
-                        zone
-                      else
-                        Spree::Country.find_by(iso: request_iso_code)&.product_zones&.active&.first
-                      end
-                    end
+    return unless flow_io_zones.present?
```

Comment:
Just came to me that we could improve the check, not instantiating the zones - we don't need the 7,8,9 lines:

```suggestion
    return unless Spree::Zones::Product.active.exists?("meta -> 'flow_data' ->> 'country' = ?", ISO3166::Country[request_iso_code]&.alpha3)
```

Reply:
Yes!!! I will make this change as well as the ones you mentioned in https://github.com/mejuri-inc/mejuri-web/pull/741, test them out and if everything works ok I will merge both PRs! we will eventually review this hehe!